src/Common/src/System/Security/Cryptography/RSAOpenSsl.cs

```diff
@@ -156,20 +209,25 @@ public override byte[] Encrypt(byte[] data, RSAEncryptionPadding padding)
             if (padding == null)
                 throw new ArgumentNullException(nameof(padding));
 
-            Interop.Crypto.RsaPadding rsaPadding = GetInteropPadding(padding);
+            Interop.Crypto.RsaPadding rsaPadding = GetInteropPadding(padding, out RsaPaddingProcessor oaepProcessor);
             SafeRsaHandle key = _key.Value;
             CheckInvalidKey(key);
 
             byte[] buf = new byte[Interop.Crypto.RsaSize(key)];
 
-            int returnValue = Interop.Crypto.RsaPublicEncrypt(
-                data.Length,
+            bool encrypted = TryEncrypt(
+                key,
                 data,
                 buf,
-                key,
-                rsaPadding);
+                rsaPadding,
+                oaepProcessor,
+                out int bytesWritten);
 
-            CheckReturn(returnValue);
+            if (!encrypted || bytesWritten != buf.Length)
+            {
+                Debug.Fail("TryEncrypt behaved unexpectedly");
```

Comment:
Since there are multiple conditions that could have led to this failure, consider putting details into the assert, e.g.
```C#
Debug.Fail($"TryEncrypt behaved unexpectedly: {nameof(encrypt)}=={encrypted}, {nameof(bytesWritten)}=={bytesWritten}, {nameof(buf.Length)}=={buf.Length}");
```